mne/export/tests/test_export.py

```diff
@@ -46,6 +51,96 @@ def test_export_raw_eeglab(tmpdir):
     assert_allclose(raw.get_data(), raw_read.get_data())
 
 
+@pytest.mark.skipif(not _check_edflib_installed(strict=False),
+                    reason='edflib-python not installed')
+@pytest.mark.parametrize('format', ['edf', 'bdf'])
+def test_integer_sfreq_edf_and_bdf(tmp_path, format):
+    """Test saving a Raw array with integer sfreq to EDF and BDF."""
+    info = create_info(['1', '2', '3'], sfreq=1000, ch_types='eeg')
+    data = np.random.random(size=(3, 1000))
+    raw = RawArray(data, info)
+
+    temp_fname = op.join(str(tmp_path), f'test.{format}')
+
+    raw.export(temp_fname)
+    if format == 'edf':
+        raw_read = read_raw_edf(temp_fname, preload=True)
+    elif format == 'bdf':
+        raw_read = read_raw_bdf(temp_fname, preload=True)
+
+    assert raw.ch_names == raw_read.ch_names
+    # only compare the original length, since extra zeros are appended
+    orig_raw_len = len(raw)
+    assert_array_almost_equal(
+        raw.get_data(), raw_read.get_data()[:, :orig_raw_len], decimal=3)
```

Comment:
```suggestion
        raw.get_data(), raw_read.get_data()[:, :orig_raw_len], decimal=4)
```
works on my machine.
Please also put a comment on why it's so high tolerance. I assume it's due to int16 quantisation?

Reply:
Added comment.